testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -59,9 +63,119 @@ pub struct BitcoinRegtestController {
     chain_tip: Option<BurnchainTip>,
     use_coordinator: Option<CoordinatorChannels>,
     burnchain_config: Option<Burnchain>,
-    last_utxos: Vec<UTXO>,
-    last_tx_len: u64,
-    min_relay_fee: u64, // satoshis/byte
+    ongoing_block_commit: Option<OngoingBlockCommit>,
+}
+
+struct OngoingBlockCommit {
+    payload: LeaderBlockCommitOp,
+    utxos: UTXOSet,
+    fees: LeaderBlockCommitFees,
+    txids: Vec<Txid>,
+}
+
+impl OngoingBlockCommit {
+    fn sum_utxos(&self) -> u64 {
+        self.utxos.total_available()
+    }
+}
+
+#[derive(Clone)]
+struct LeaderBlockCommitFees {
```

Comment:
Why are any of these fields important to track here?  The _tenure processor_ decides how much to commit and to burn.  This code should only be concerned with the Bitcoin transaction fees (the fee for the whole transaction, and any state required to replace-by-fee).

Reply:
Is the concern being about having

```rust
struct Fee {
   fee_1: u64,
   fee_2: u64,
   fee_3: u64,
} 

impl Fee {
   fn get_total(&self) -> u64 {
       fee_1 + fee_2 + fee_3 
   }
}
```

vs 

```rust
struct Fee {
   total: u64,
} 

impl Fee {
   fn new(fee_1: u64, fee_2: u64, fee_3: u64) -> Fee {
       Fee {
           total: fee_1 + fee_2 + fee_3
       }
   }
}
```
?

Reply:
The concern is about why `sunset_fee`, `outputs_len`, and `sortition_fee` are even listed, as well as `amount_per_output()`.  The code shouldn't care about how the block-commit was structured, nor should it be trying to (re-)create the block-commit's outputs.  The purpose of this module is to provide a caller a way to construct a Bitcoin transaction, and then to attempt to send it.  It shouldn't be surreptitiously re-building it, because it can't know the caller's intentions.

Reply:
We are clearly missing an abstraction, before this iteration, the code used to look like this:

```rust
fn build_leader_block_commit_tx(
        ...
        let number_of_transfers = payload.commit_outs.len() as u64;		         
        let value_per_transfer = payload.burn_fee / number_of_transfers;
        ...
```

It should not be up to the `build_leader_block_commit_tx` method to manipulate payload's fields. We should either be associating some additional methods to `payload`, or abstracting this somehow. This PR is introducing a new struct for centralizing the leader commit operation costs structure needed for RBF and it felt natural to add some helpers in there.